xb.bat

```diff
@@ -29,12 +29,27 @@ REM Utilities
 REM ============================================================================
 
 :check_python
-SETLOCAL
+setlocal enabledelayedexpansion
 SET FOUND_PYTHON_EXE=""
-1>NUL 2>NUL CMD /c where python3
-IF NOT ERRORLEVEL 1 (
-  SET FOUND_PYTHON_EXE=python3
+FOR /F %%I IN ('where python') DO (
+	CMD /C %%I -c "import sys; sys.exit(1 if not sys.version_info[:2] >= (3, 4) else 0)"
+	IF ERRORLEVEL 0 (
+	  SET _RESULT=0
+	  SET PYTHON_EXE=%%I
+	  ENDLOCAL enabledelayedexpansion
+	  GOTO :eof
+	)
 )
+FOR /F %%I IN ('where python3') DO (
+	CMD /C %%I -c "import sys; sys.exit(1 if not sys.version_info[:2] >= (3, 4) else 0)"
```

Comment:
Not awfully familiar with bash, but maybe we can abstract this into its own function just so the check isn't duplicated across the two executables. Something like `check_python_version 3` or such.

Reply:
Not really sure what you mean here. The check won't be duplicated. It'll check all python3.exe's in PATH and then all python.exe's in PATH

Reply:
Yeah - I'm just referring to code duplication. Though this is batch, so I'm not really sure if functions are that elegant either.

Reply:
I'm not too familiar with batch either. There might be a more elegant solution but I'm not aware of it. Well..I supposed I could create another goto and have it work that way but that seems equally as bad.

Reply:
Yeah, that's alright - not a big deal